src/Microsoft.Diagnostics.Runtime/src/DataReaders/Core/CoreDumpReader.cs

```diff
@@ -150,39 +151,35 @@ public bool Read(ulong address, Span<byte> buffer, out int bytesRead)
             return bytesRead > 0;
         }
 
-        public ulong ReadPointer(ulong addr)
+        public ulong ReadPointer(ulong address)
         {
-            Span<byte> buffer = stackalloc byte[IntPtr.Size];
-
-            if (_core.ReadMemory((long)addr, buffer) == IntPtr.Size)
-                return buffer.AsPointer();
-
-            return 0;
+            ReadPointer(address, out ulong value);
+            return value;
         }
 
-        public unsafe bool Read<T>(ulong addr, out T value) where T : unmanaged
+        public unsafe bool Read<T>(ulong address, out T value) where T : unmanaged
         {
             Span<byte> buffer = stackalloc byte[sizeof(T)];
-            if (!Read(addr, buffer, out _))
+            if (Read(address, buffer, out int size) && size == sizeof(T))
             {
-                value = Unsafe.As<byte, T>(ref buffer[0]);
+                value = Unsafe.As<byte, T>(ref MemoryMarshal.GetReference(buffer));
```

Comment:
Ah right.  Thanks for these cleanups.  I'm still getting used to Unsafe.As and MemoryMarshal after being stuck on an older version of Desktop .Net for years.